mainauto.c

```diff
@@ -0,0 +1,39 @@
+#include <stdio.h>
+#include "display.h"
+#include "mapio.h"
+#include "datastrc.h"
+#include "control.h"
+
+int main(int argc, char &argv){
```

Comment:
You've got the pointer symbols the other way around.
Should be `char *argv[]` as the argument that this function takes.

`*something` means: "object that is pointed to by `something`". So you use `*` when you **have a pointer** and want to **get the value**.

`&something` means: "address of `something`". So you use `&` when you **have a value** and want to **get a pointer**.